frontend/components/commons/sidebar-feedback-task/SidebarFeedbackTask.vue

```diff
@@ -0,0 +1,100 @@
+<!--
+  - coding=utf-8
+  - Copyright 2021-present, the Recognai S.L. team.
+  -
+  - Licensed under the Apache License, Version 2.0 (the "License");
+  - you may not use this file except in compliance with the License.
+  - You may obtain a copy of the License at
+  -
+  -     http://www.apache.org/licenses/LICENSE-2.0
+  -
+  - Unless required by applicable law or agreed to in writing, software
+  - distributed under the License is distributed on an "AS IS" BASIS,
+  - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  - See the License for the specific language governing permissions and
+  - limitations under the License.
+  -->
+
+<template>
+  <div class="sidebar" :class="expandedComponent ? '--expanded' : null">
+    <SidebarFeedbackTaskButtonGroup
+      v-for="group in sidebarGroups"
+      :key="group"
+      :groupName="group"
+      :groupItems="filteredSidebarItemsByGroup(group)"
+      :active-buttons="activeButtons"
+      @on-click-sidebar-action="onClickSidebarAction"
+    />
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    sidebarItems: {
+      type: Array,
+      required: true,
+    },
+    activeButtons: {
+      type: Array,
+      required: true,
+    },
+    expandedComponent: {
+      type: String,
+      default: null,
+    },
+  },
+  computed: {
+    sidebarGroups() {
+      const groups = [
+        ...new Set(this.sidebarItems.map((button) => button.group)),
+      ];
+      return groups;
+    },
+  },
```

Comment:
if you pass a sidebarObject (where keys are the mode) instead of a list sidebarItems, you will have better performance since it's not necesary to compute the sidebarGroups anymore